### PR TITLE
ChatLog

### PR DESCRIPTION
**Pull request by MateiPribagu**

### CODE REVIEW

**Comment by MateiPribagu** [comment]
Removed the sky passage garrison flags when your turn ends.
Fixed bug where playing a Play: Add card to hand when you had ten cards would not add the card to your hand.
Fixed bug where fleeting cards would leave behind sky passage garrison flag.

File: match.c [hunk 1]
```diff
@@ -307,17 +311,20 @@ inactive
 		for (x=yGetDatabaseCount("p"+p+"hand"); >0) {
 			yDatabaseNext("p"+p+"hand");
 			if (HasKeyword(FLEETING, 1*yGetVar("p"+p+"hand", "keywords"))) {
+				trUnitSelectClear();
+				trUnitSelectByID(1*yGetVar("p"+p+"hand", "pos"));
```
Comment (jkalexho220) [inline match.c:315]: this doesn't do anything lmao

File: match.c [hunk 2]
```diff
@@ -307,17 +311,20 @@ inactive
...
 				trUnitChangeProtoUnit("Hero Death");
 				if (trCurrentPlayer() == p) {
-					if (yGetVar("p"+p+"hand", "spell") == SPELL_NONE) {
-						type = yGetVar("p"+p+"hand", "proto");
-						trChatSend(0, "Discarded a " + trStringQuestVarGet("card_" + type + "_name"));
-					} else {
-						type = yGetVar("p"+p+"hand", "spell");
-						trChatSend(0, "Discarded a " + trStringQuestVarGet("spell_" + type + "_name"));
-					}
+					trSoundPlayFN("olympustemplesfx.wav","1",-1,"","");
```
Comment (jkalexho220) [inline match.c:320]: set a qv instead and play sound afterwards

File: shared.c [hunk 3]
```diff
@@ -498,6 +498,30 @@ void DeploySober(string p="", string v=""){
...
+	if (trCurrentPlayer() == p) {
+		trChatHistoryClear();
+	}
+	int x = peekModularCounterNext("chat" + p + "Log");
+	for(i=x;<11){
+		trChatSendToPlayer(0, p, trStringQuestVarGet("chat" + p + "Log" + i));
+	}
+	for(i=1;<x){
+		trChatSendToPlayer(0, p, trStringQuestVarGet("chat" + p + "Log" + i));
+	}
```
Comment (jkalexho220) [inline shared.c:514]: for (i =  12; >0) {
        trChatSendToPlayer(0, p, trStringQuestVarGet("chat" + p + "Log" + modularCounterNext("chat"+p+"log")));
}

**Pull request merged**